### PR TITLE
Be resilient to having a project that doesn't know about ArgumentNullException or ArgumentException

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/19171

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide 

Note: i couldn't find a way to test this.  I did manually validate it though with the scenario dave referenced.

File: src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -369,11 +374,25 @@ private bool IsNullLiteral(IOperation operand)
...
+        private static SyntaxNode GetTypeNode(
+            Compilation compilation, SyntaxGenerator generator, Type type)
+        {
+            var typeSymbol = compilation.GetTypeByMetadataName(type.FullName);
+            if (typeSymbol == null)
+            {
+                return generator.QualifiedName(
+                    generator.IdentifierName(nameof(System)),
+                    generator.IdentifierName(type.Name));
+            }
```
Comment (Pilchie) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: Why not `SyntaxFactory.ParseTypeName(type.FullName)` here, instead of throwing away the fullname part?

File: src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs [hunk 2]
```diff
@@ -704,5 +704,56 @@ public C(string s)
...
+        {
+            throw new System.Exception();
+        }
+    }
+}");
+        }
+
+        [WorkItem(19174, "https://github.com/dotnet/roslyn/issues/19174")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInitializeParameter)]
+        public async Task TestRespectPredefinedTypePreferences()
```
Comment (Pilchie) [inline src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs]: this test looks like it belongs in another PR.

File: src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs [hunk 3]
```diff
@@ -117,7 +117,10 @@ private bool IsIfNullCheck(IOperation statement, IParameterSymbol parameter)
         {
             if (statement is IIfStatement ifStatement)
             {
-                if (ifStatement.Condition is IBinaryOperatorExpression binaryOperator)
+                var condition = ifStatement.Condition;
+                condition = UnwrapImplicitConversion(condition);
```
Comment (Pilchie) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: What is this for?  Is there a bug that we should have a test for here?

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 

**Customer scenario**

This fixes a crash in "add parameter null check" when in a project that didn't know about the types ArgumentException or ArgumentNullException

**Bugs this fixes:**

Fixes #19174

**Workarounds, if any**

User adds reference to dlls that contain these types first.

**Risk**

Low.

**Performance impact**

Low.

**Is this a regression from a previous update?**

No.  New feature.

**Root cause analysis:**

We don't generally test against projects that are missing core dll references.

**How was the bug found?**

Customer report.

File: src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: Seemed unnecessary to do parsing (plus, i'm just not a fan of codegen code actually doing parsing".  I already know the name of hte type, so i can just directly create the final syntax nodes we want.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_vs-integration_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

File: src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs (hunk 1 above)
Comment (Pilchie) [inline src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs]: It just seems strange to go to the trouble of using type.FullName in most of the codepaths, but then ignore it and hard-code something else instead.
